specification/iothub/resource-manager/Microsoft.Devices/preview/2021-02-01-preview/iothub.json

```diff
@@ -4240,8 +4241,18 @@
           "readOnly": true
         },
         "type": {
-          "description": "Identity type. Only allowed values are SystemAssigned and UserAssigned. Comma separated if both for ex: SystemAssigned,UserAssigned",
-          "type": "string"
+          "description": "The type of identity used for the resource. The type 'SystemAssigned,UserAssigned' includes both an implicitly created identity and a set of user assigned identities. The type 'None' will remove any identities from the service.",
+          "type": "string",
+          "enum": [
+            "SystemAssigned",
+            "UserAssigned",
+            "SystemAssigned,UserAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": true
```

Comment:
modelAsString should be set to false as mentioned in above example.

Reply:
@RamyasreeChakka we are setting modelAsString : true, as other enums in our swagger were using in that way and to maintain that consistent behaviour.

Reply:
@nimengan - ARM common definitions are present here: https://github.com/Azure/azure-rest-api-specs/blob/a06d912bc912010bdc0aa5d1d3c4b28ee1d89905/specification/common-types/resource-management/v2/types.json#L221
These are the guidance for RPs. In the common definition also, for identity -> modelAsString is set to false to allow strict validation. I would leave it up to you make the call here. Signing off from ARM side.

For more details: http://azure.github.io/autorest/extensions/#x-ms-enum